lib/BuildSystem/BuildSystem.cpp

```diff
@@ -885,6 +887,123 @@ class DirectoryContentsTask : public Task {
 };
 
 
+/// This task is responsible for computing the filtered lists of files in
+/// directories.
+class FilteredDirectoryContentsTask : public Task {
+  std::string path;
+
+  /// The exclusion filters used while computing the signature
+  StringList filters;
+
+  /// The value for the input directory.
+  BuildValue directoryValue;
+
+  virtual void start(BuildEngine& engine) override {
+    // Depend on the node so that it can be connected up to potential producers.
+    engine.taskNeedsInput(
+        this, BuildKey::makeNode(path).toData(), /*inputID=*/0);
+    // Depend on the raw stat information, because we want to know if we need
+    // to rescan items under this if anything has changed.
+    engine.taskNeedsInput(
+        this, BuildKey::makeStat(path).toData(), /*inputID=*/1);
+  }
+
+  virtual void providePriorValue(BuildEngine&,
+                                 const ValueType& value) override {
+  }
+
+  virtual void provideValue(BuildEngine&, uintptr_t inputID,
+                            const ValueType& value) override {
+    // If our value is uninitialized, take whatever build value came in
```

Comment:
Hmmm, I'm not sure about this. Why wouldn't we just always take the directoryValue from `inputID == 1`?

Reply:
Yeah, this was the point I was most interested in. I think, based on our discussion of the serial nature of the dependencies, that inputID == 1 should be sufficient.